extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/TransactionManagerConfiguration.java

```diff
@@ -16,10 +16,24 @@
 public final class TransactionManagerConfiguration {
     /**
      * The node name used by the transaction manager.
+     * Must not exceed a length of 28 bytes.
+     *
+     * @see #shortenNodeNameIfNecessary
      */
     @ConfigItem(defaultValue = "quarkus")
     public String nodeName;
 
+    /**
+     * Whether the node name should be shortened if necessary.
+     * The node name must not exceed a length of 28 bytes. If this property is set to {@code true}, and the node name exceeds 28
+     * bytes, the node name is shortened by calculating the <a href="https://en.wikipedia.org/wiki/SHA-2">SHA-224</a> hash,
+     * which has a length of 28 bytes.
+     *
+     * @see #nodeName
+     */
+    @ConfigItem(defaultValue = "false")
+    public boolean shortenNodeNameIfNecessary;
```

Comment:
How about naming it simply `truncateNodeName`? The `ifNecessary` makes my chin itch a bit 😀

Reply:
How about making it true by default as a precaution?

Reply:
this is not truncation. Truncate is rather abrupt/finite in nature.

maybe compress as this will be a lossy-but-hopefully-not-significant-compression ?


btw. why the "ifNecessary" part ? if its necessary why is it not true by default?

Reply:
i.e. since this seems necessary I would argue it should be default on and possible to turn off. At least in main with good release note metnion - if this was for LTS branch I could see how one could argue it should stay off.

Am I missing something?

Reply:
> this is not truncation. Truncate is rather abrupt/finite in nature.
> 
> maybe compress as this will be a lossy-but-hopefully-not-significant-compression ?

Right, `compress` makes more sense

> 
> btw. why the "ifNecessary" part ? if its necessary why is it not true by default?

+1

Reply:
Maybe it's less unlikely to happen, but would it truncating be a problem if the algorithm isn't available?

Reply:
> How about making it true by default as a precaution?

@gastaldi https://github.com/quarkusio/quarkus/pull/36752#pullrequestreview-1703877795

Reply:
> Maybe it's less unlikely to happen, but would truncating be a problem if the algorithm isn't available?

Hard to say. This depends how the node names are constructed. If one has long names and a common prefix, then this might lead to name duplication.

Reply:
> > btw. why the "ifNecessary" part ? if its necessary why is it not true by default?

Historically the default is to fail startup if the nodeIdentifier, set by the user or the platform, is too long. Since the use of a hash incurs the risk of collisions and can result in non ACID behaviour it should not be the default - the user has to opt in to the added risk of using a hash since only they can quantify the risk to their application.

Reply:
ok, how about simply "hashNodeNameIfNecessary" or simply "hashNodeName" which still will only happen if needed but with hash it at least signals what happens and collision awareness could be necessary.

Reply:
Sounds reasonable.